spec/tools/launcher/launcher_config_spec.rb

```diff
@@ -0,0 +1,460 @@
+          tf.unlink
+        end
+
+        it "raises with a parameter" do
+          tf = Tempfile.new('mylauncher.txt')
+          tf.puts "DIVIDER 1"
+          tf.close
+
+          expect { LauncherConfig.new(tf.path) }.to raise_error(Cosmos::ConfigParser::Error)
+          tf.unlink
+        end
+      end
+
+      describe 'NUM_COLUMNS' do
+        it "parses a single value" do
+          tf = Tempfile.new('mylauncher.txt')
+          tf.puts "NUM_COLUMNS 6"
+          tf.close
+
+          lc = LauncherConfig.new(tf.path)
+          expect(lc.num_columns).to eq 6
+          tf.unlink
+        end
+
+        it "raises with a string" do
+          tf = Tempfile.new('mylauncher.txt')
+          tf.puts "NUM_COLUMNS TWO"
+          tf.close
+
+          expect { LauncherConfig.new(tf.path) }.to raise_error(Cosmos::ConfigParser::Error)
+          tf.unlink
+        end
+
+        it "raises with more than 1 parameter" do
+          tf = Tempfile.new('mylauncher.txt')
+          tf.puts "NUM_COLUMNS 2 3"
+          tf.close
+
+          expect { LauncherConfig.new(tf.path) }.to raise_error(Cosmos::ConfigParser::Error)
+          tf.unlink
+        end
+      end
+
+      describe 'LABEL' do
+        it "parses a single value" do
+          tf = Tempfile.new('mylauncher.txt')
+          tf.puts "LABEL 'HI THERE'"
+          tf.close
+
+          lc = LauncherConfig.new(tf.path)
+          expect(lc.items[0][0]).to eq :LABEL
+          expect(lc.items[0][1]).to eq "HI THERE"
+          expect(lc.items[0][2]).to be_nil
+          expect(lc.items[0][3]).to be_nil
+          expect(lc.items[0][4]).to be_nil
+          tf.unlink
+        end
+
+        it "raises with more than 1 parameter" do
+          tf = Tempfile.new('mylauncher.txt')
+          tf.puts "LABEL HI THERE"
+          tf.close
+
+          expect { LauncherConfig.new(tf.path) }.to raise_error(Cosmos::ConfigParser::Error)
+          tf.unlink
+        end
+      end
+
+      describe 'TOOL_FONT' do
+        it "parses the font and size" do
+          tf = Tempfile.new('mylauncher.txt')
+          tf.puts "TOOL_FONT tahoma 12"
+          tf.close
+
+          lc = LauncherConfig.new(tf.path)
+          expect(lc.tool_font_settings[0]).to eq 'tahoma'
+          expect(lc.tool_font_settings[1]).to eq 12
+          tf.unlink
+        end
+
+        it "raises with only 1 parameter" do
+          tf = Tempfile.new('mylauncher.txt')
+          tf.puts "TOOL_FONT tahoma"
+          tf.close
+
+          expect { LauncherConfig.new(tf.path) }.to raise_error(Cosmos::ConfigParser::Error)
+          tf.unlink
+        end
+
+        it "raises with more than 2 parameters" do
+          tf = Tempfile.new('mylauncher.txt')
+          tf.puts "TOOL_FONT tahoma 12 16"
+          tf.close
+
+          expect { LauncherConfig.new(tf.path) }.to raise_error(Cosmos::ConfigParser::Error)
+          tf.unlink
+        end
+
+        # TODO: Is it possible to check for bad fonts?
```

Comment:
We could do a lookup on the fonts maybe but I didn't know how this might work cross platform.